torchtune/utils/precision.py

```diff
@@ -161,18 +171,20 @@ def set_default_dtype(dtype: torch.dtype) -> Generator[None, None, None]:
         torch.set_default_dtype(old_dtype)
 
 
-def validate_expected_param_dtype(model: torch.nn.Module, dtype: torch.dtype) -> None:
+def validate_expected_param_dtype(
+    named_params: Iterable[Tuple[str, nn.Parameter]], dtype: torch.dtype
```

Comment:
nit: technically the get_adapter_params method in peft_utils is typed as returning a `Dict[str, Any]` that we pass along to this, so if we're doing it this way we may want to change `Any` -> `nn.Parameter`